server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -107,6 +108,21 @@ public class Metadata implements Iterable<IndexMetadata>, Diffable<Metadata>, To
     public static final String UNKNOWN_CLUSTER_UUID = Strings.UNKNOWN_UUID_VALUE;
     public static final Pattern NUMBER_PATTERN = Pattern.compile("[0-9]+$");
 
+    /**
+     * Utility to identify whether input index uses SEGMENT replication strategy in established cluster state metadata.
+     *
+     * @param indexName Index name
+     * @return true if index uses SEGMENT replication, false otherwise
+     */
+    public boolean isSegmentReplicationEnabled(String indexName) {
```

Comment:
nit: This utitlity will now be consumed by different plugins. Do we need to add a note stating this ? Even though this utility is public (and non-internal), having a explicit comment ensures future changes are done cautiously.

Reply:
@dreamer-89 I don't think this makes sense.  This class is labelled [@opensearch.internal ](https://github.com/opensearch-project/OpenSearch/blob/0b2c07a60252292a5a2e41e3af17412ed940886d/DEVELOPER_GUIDE.md#L576-L575)and is not intended to be extended nor does it guarantee bwc.

Reply:
Thanks @mch2 for the comment. Yes, I agree internal classs is not intended for external usage but in current state we already have `ClusterState` used inside multiple plugins and this utility provide mechanism to identify segrep enabled indices. For same reasons, we added it in the first place via https://github.com/opensearch-project/OpenSearch/issues/9566